File: code_challenge_solution/java/66 Fibonacci sequence.java
public class FibonacciSequence {
    public static void main(String[] args) {
        int n = 20; // define the number of terms to generate in the sequence
        
        // variables to store the current and previous terms in the sequence
        int currentTerm = 0;
        int previousTerm = 1;
        
        // print the first two terms in the sequence
        System.out.print(previousTerm + " " + currentTerm + " ");
        
        // generate the remaining terms in the sequence
        for (int i = 2; i < n; i++) {
            int nextTerm = previousTerm + currentTerm;
            System.out.print(nextTerm + " ");
            
            previousTerm = currentTerm;
            currentTerm = nextTerm;
        }
    }
}

File: code_challenge_solution/c++/66 Fibonacci sequence.cpp
#include <iostream>

int main() {
    int n;
    std::cout << "Enter the number of terms in the Fibonacci sequence: ";
    std::cin >> n;

    int firstTerm = 0, secondTerm = 1, nextTerm;

    std::cout << "Fibonacci sequence up to " << n << " terms: ";
    std::cout << firstTerm << " " << secondTerm << " ";

    for (int i = 3; i <= n; ++i) {
        nextTerm = firstTerm + secondTerm;
        std::cout << nextTerm << " ";
        firstTerm = secondTerm;
        secondTerm = nextTerm;
    }

    return 0;
}